### ISSUE TITLE
[sig-storage] HostPath should support existing directory subPath flake

### ISSUE THREAD

**Issue by kargakis** [openshift/origin#18823]
```
/tmp/openshift/build-rpms/rpm/BUILD/origin-3.10.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/common/host_path.go:121
Expected error:
    <*errors.errorString | 0xc420c86440>: {
        s: "expected pod \"pod-host-path-test\" success: pod \"pod-host-path-test\" failed with status: {Phase:Failed Conditions:[] Message:Pod Predicate MatchNodeSelector failed Reason:MatchNodeSelector HostIP: PodIP: StartTime:2018-03-04 07:18:54 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[] QOSClass:}",
    }
    expected pod "pod-host-path-test" success: pod "pod-host-path-test" failed with status: {Phase:Failed Conditions:[] Message:Pod Predicate MatchNodeSelector failed Reason:MatchNodeSelector HostIP: PodIP: StartTime:2018-03-04 07:18:54 +0000 UTC InitContainerStatuses:[] ContainerStatuses:[] QOSClass:}
not to have occurred
/tmp/openshift/build-rpms/rpm/BUILD/origin-3.10.0/_output/local/go/src/github.com/openshift/origin/vendor/k8s.io/kubernetes/test/e2e/framework/util.go:2197
```
https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/batch/test_pull_request_origin_extended_conformance_gce/17025/
@openshift/sig-storage 
/kind test-flake

**Comment by deads2k** [comment]
This isn't flaking.  It is 100% failures

**Comment by deads2k** [comment]
100% failures after March 4 midnight https://deck-ci.svc.ci.openshift.org/?job=test_pull_request_origin_extended_conformance_gce

**Comment by deads2k** [comment]
https://deck-ci.svc.ci.openshift.org/?job=test_branch_origin_extended_conformance_gce indicates that 3.9 is also affected.  This must be related to infrastructure or job definition.  The 3.9 commit has nothing to do with this area.

**Comment by kargakis** [comment]
@smarterclayton

**Comment by derekwaynecarr** [comment]
the failure does not appear contianer runtime related... the pod does not appear to ever schedule due to failing match node selectors.  

```
Mar  4 07:18:56.715: INFO: Pod "pod-host-path-test": Phase="Failed", Reason="MatchNodeSelector", readiness=false. Elapsed: 2.18721399s
```

@aveshagarwal -- can you dig deeper on this?

**Comment by aveshagarwal** [comment]
@derekwaynecarr sure looking.

**Comment by aveshagarwal** [comment]
The node where this pod is being scheduled has infra label:

map[string]string{beta.kubernetes.io/arch: amd64,beta.kubernetes.io/instance-type: n1-standard-2,beta.kubernetes.io/os: linux,failure-domain.beta.kubernetes.io/region: us-east1,failure-domain.beta.kubernetes.io/zone: us-east1-c,kubernetes.io/hostname: ci-prtest-5a37c28-17025-ig-m-hdt0,node-role.kubernetes.io/infra: true,node-role.kubernetes.io/master: true,role: infra,

Is it expected this test pod to get scheduled on the infra node?

Or the test is incorrectly selecting infra node?

Still checking what is going on if the above is really the issue.

**Comment by derekwaynecarr** [comment]
working on an upstream pr to fix the test to let the scheduler schedule the pod rather than self scheduling.  this would let it work fine with any admission controllers running.

**Comment by derekwaynecarr** [comment]
hmm, of course, this change is not as trivial given the test structure.

the test self-schedules pods, but prior to the pod actually being scheduled or run, it creates a directory on the machine it intends to run the pod against to validate that the pod sees that subdir.  as a result, changing the test is non-obvious...

**Comment by derekwaynecarr** [comment]
simpler option for now is to disable namespace node selecting for upstream tests.

see: https://github.com/openshift/origin/blob/master/test/extended/util/test.go#L231-L234

given how frequently this clause is used that would probably work best until we can change upstream.

**Comment by aveshagarwal** [comment]
Just for reference, https://github.com/openshift/origin/pull/18816 should address this.
@derekwaynecarr

**Comment by jpeeler** [comment]
This has been fixed by #18842 and #18843.
